tests/foreman/api/test_errata.py

```diff
@@ -53,866 +53,896 @@
 from robottelo.helpers import add_remote_execution_ssh_key
 from robottelo.products import RepositoryCollection
 from robottelo.products import YumRepository
-from robottelo.test import APITestCase
 from robottelo.vm import VirtualMachine
 
 CUSTOM_REPO_URL = FAKE_9_YUM_REPO
 CUSTOM_REPO_ERRATA_ID = FAKE_2_ERRATA_ID
 
 
+@skip_if_not_set('clients', 'fake_manifest')
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
 @pytest.mark.run_in_one_thread
-class ErrataTestCase(APITestCase):
+class TestErrata:
     """API Tests for the errata management feature"""
 
-    @classmethod
-    @skip_if_not_set('clients', 'fake_manifest')
-    @pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
-    def setUpClass(cls):
-        """Create Org, Lifecycle Environment, Content View, Activation key"""
-        super().setUpClass()
-        cls.org = entities.Organization().create()
-        cls.env = entities.LifecycleEnvironment(organization=cls.org).create()
-        cls.content_view = entities.ContentView(organization=cls.org).create()
-        cls.activation_key = entities.ActivationKey(
-            environment=cls.env, organization=cls.org
-        ).create()
-        setup_org_for_a_rh_repo(
-            {
-                'product': PRDS['rhel'],
-                'repository-set': REPOSET['rhst7'],
-                'repository': REPOS['rhst7']['name'],
-                'organization-id': cls.org.id,
-                'content-view-id': cls.content_view.id,
-                'lifecycle-environment-id': cls.env.id,
-                'activationkey-id': cls.activation_key.id,
-            },
-            force_manifest_upload=True,
-        )
-        cls.custom_entities = setup_org_for_a_custom_repo(
-            {
-                'url': FAKE_9_YUM_REPO,
-                'organization-id': cls.org.id,
-                'content-view-id': cls.content_view.id,
-                'lifecycle-environment-id': cls.env.id,
-                'activationkey-id': cls.activation_key.id,
+
+@pytest.fixture(scope='module')
+def activation_key(module_org, module_lce):
+    activation_key = entities.ActivationKey(
+        environment=module_lce, organization=module_org
+    ).create()
+    return activation_key
+
```

Comment:
@devendra104 I do not want to use `yield` because I do not want my tests wasting time uninstalling things when the entire satellite will be erased shortly after the test. Doing so would just impose unnecessary load on the satellite (and in the case of orgs, lead to race conditions).